python/pyarrow/_plasma.pyx

```diff
@@ -312,6 +325,10 @@ cdef class PlasmaClient(_Weakrefable):
         self.notification_fd = -1
         self.store_socket_name = b""
 
+        warnings.warn(
+            "Plasma is deprecated since Arrow 10.0.0. It will be removed in 12.0.0 or so.",
+            DeprecationWarning)
```

Comment:
```suggestion
            DeprecationWarning, stacklevel=3)
```

(taking one more so that it works better for `connect(..)`, alternative would be to move this warning to connect() itself)